scripts/process_data.py

```diff
@@ -0,0 +1,234 @@
+
+# coding: utf-8
+
+# # Processing all datasets to be used in downstream analyses
+
+# In[1]:
+
+import os
+import requests
+import numpy as np
+import pandas as pd
+
+from sklearn import preprocessing
+
+
+# ## Define Input and Output Filenames
+
+# In[2]:
+
+# Input Files
+rna_file = os.path.join('data', 'raw', 'HiSeqV2')
+mut_file = os.path.join('data', 'raw', 'PANCAN_mutation')
+copy_file = os.path.join('data', 'raw', 'Gistic2_CopyNumber_Gistic2_all_thresholded.by_genes')
+clinical_file = os.path.join('data', 'raw', 'samples.tsv')
+
+
+# In[3]:
+
+# Output Files
+# Processing RNAseq data by z-score and zeroone norm
+rna_out_file = os.path.join('data', 'pancan_scaled_rnaseq.tsv')
+rna_out_zeroone_file = os.path.join('data', 'pancan_scaled_zeroone_rnaseq.tsv')
+
+# Mutation Data
+mut_out_file = os.path.join('data', 'pancan_mutation.tsv')
+
+# Two copy number matrices, for thresholded (2) gains and losses
+copy_gain_out_file = os.path.join('data', 'copy_number_gain.tsv')
+copy_loss_out_file = os.path.join('data', 'copy_number_loss.tsv')
+
+# Clinical data
+clinical_processed_out_file = os.path.join('data', 'clinical_data.tsv')
+
+# OncoKB output file
+oncokb_out_file = os.path.join('data', 'oncokb_genetypes.tsv')
+
+# Status matirix integrating mutation and copy number events
+known_status_file = os.path.join('data', 'status_matrix.tsv')
+
+
+# ## Load Data
+
+# In[4]:
+
+rnaseq_df = pd.read_table(rna_file, index_col=0)
+mutation_df = pd.read_table(mut_file)
+copy_df = pd.read_table(copy_file, index_col=0)
+clinical_df = pd.read_table(clinical_file, index_col=0, low_memory=False)
+
+
+# ## Begin processing different data types
+# 
+# ### RNAseq
+
+# In[5]:
+
+# Process RNAseq file
+rnaseq_df.index = rnaseq_df.index.map(lambda x: x.split('|')[0])
+rnaseq_df.columns = rnaseq_df.columns.str.slice(start=0, stop=15)
+rnaseq_df = rnaseq_df.drop('?').fillna(0).sort_index(axis=1)
+
+# Gene is listed twice in RNAseq data, drop both occurrences
+rnaseq_df.drop('SLC35E2', axis=0, inplace=True)
```

Comment:
I tried `drop_duplicates()` but was getting this error `RecursionError: maximum recursion depth exceeded`. It will be faster to drop the genes I know have specific problems. `SLC35E2` is the only one with this issue in the USCS Xena data.